src/fontra/views/editor/scene-controller.js

```diff
@@ -1148,6 +1168,49 @@ export class SceneController {
     });
   }
 
+  async closeContour() {
+    const openContours = this.contextMenuState.openContourSelection;
+    const curveType = this.experimentalFeatures.quadPenTool ? "quad" : "cubic";
+    await this.editLayersAndRecordChanges((layerGlyphs) => {
+      for (const layerGlyph of Object.values(layerGlyphs)) {
+        const path = layerGlyph.path;
+        for (const contourIndex of openContours) {
+          // close open contour
+          path.contourInfo[contourIndex].isClosed = true;
+          if (curveType === "cubic") {
+            this.closeContourEnsureCubicOffCurves(path, contourIndex);
+          }
+        }
+      }
+      return "Close Contour" + (openContours.length > 1 ? "s" : "");
+    });
+  }
+
+  closeContourEnsureCubicOffCurves(path, contourIndex) {
```

Comment:
This should be a helper function, not a method. It does not depend on any state of the `this` object, it doesn't even reference `this`.

(My general rule-of-thumb: if it doesn't use `this`, most likely it should not be a method.)

Reply:
thanks. done.